identify_monographs.py

```diff
@@ -0,0 +1,402 @@
+    marc_245 = marc_record.get_fields("245")
+    for field in marc_245:
+        if field["p"] or field["n"]:
+            return True
+    # if any 250 fields are present, it's not a monograph
+    marc_250 = marc_record.get_fields("250")
+    if marc_250:
+        return True
+    # if any 505 fields are present, it's not a monograph
+    marc_505 = marc_record.get_fields("505")
+    if marc_505:
+        return True
+    return False
+
+
+def _match_records(alma_data: list, fm_data: list) -> list:
+    """Matches Alma and FileMaker records based on MMS ID."""
+    fm_data_dict = {fm_record["inventory_no"]: fm_record for fm_record in fm_data}
+    matched_records = [
+        (alma_record, fm_data_dict[alma_record["Permanent Call Number"]])
+        for alma_record in alma_data
+        if alma_record["Permanent Call Number"] in fm_data_dict
+    ]
+    return matched_records
+
+
+def _get_full_bib_data(matched_data: list, alma_bib_file: str) -> list:
+    """Given a list of tuples of Alma data references and FileMaker data, returns a list of
+    tuples of full Alma bib data and FileMaker data."""
+    alma_bib_data = pymarc.MARCReader(open(alma_bib_file, "rb"))
+    alma_bib_dict = {record["001"].data: record for record in alma_bib_data}
+    full_data = []
+    for alma_data, fm_data in matched_data:
+        alma_record = alma_bib_dict.get(alma_data["MMS Id"])
+        full_data.append((alma_record, fm_data))
+    return full_data
+
+
+def is_monograph(alma_record: Record, fm_record: dict) -> bool:
+    """Determines whether a given record is a monograph."""
+    # first, check if the record has a production type in FileMaker
+    # if not, proceed to check the Alma record
+    if not _has_fm_production_type(fm_record):
+        # check if the Alma record has a non-monograph form
+        if _has_non_monograph_alma_form(alma_record):
+            return False
+        # check if the Alma record has a non-monograph title
+        if _has_non_monograph_alma_title(alma_record):
+            return False
+        return True
+    # if the record has a production type in FileMaker, check if it's a non-monograph type
+    if _has_non_monograph_fm_production_type(fm_record):
+        return False
+    return True
+
+
+def get_monograph_forms(alma_record: dict, fm_record: dict) -> dict:
+    """For use with monograph records only. Given an Alma record and a FileMaker record,
+    returns a dictionary of monograph forms with data sources (Filemaker or Alma) as keys.
+    """
+    forms = {}
+    if _has_fm_production_type(fm_record):
+        production_types = _get_fm_monograph_production_types(fm_record)
+        normalized_production_types = _normalize_fm_monograph_production_types(
+            production_types
+        )
+        forms["filemaker"] = normalized_production_types
+    monograph_forms = _get_alma_monograph_forms(alma_record)
+    normalized_monograph_forms = _normalize_alma_monograph_form_values(monograph_forms)
```

Comment:
Since we only want to work with the normalized forms, I would move the call to `_normalize_alma_monograph_form_values` into `_get_alma_monograph_forms`, so that it normalizes the list before returning it.